tests/publiccloud/create_aistack_env.pm

```diff
@@ -0,0 +1,276 @@
+    assert_script_run("kubectl create ns $namespace");
+    assert_script_run("kubectl create secret docker-registry application-collection --docker-server=dp.apps.rancher.io --docker-username='$docker_user_name' --docker-password='$SECRET_application_collection' -n $namespace", timeout => 120);
+
+    # Install private-ai-stack
+    my $gitlab_clone_url = 'https://git:' . $git_token . '@' . $repo_url;
+    assert_script_run("git clone $gitlab_clone_url");
+    #assert_script_run( "curl " . data_url("aistack/$vf_name") . " -o $vf_name", 60);
+    assert_script_run("curl -o $vf_name $ai_chart_repo", timeout => 120);
+
+    # local-path-storage.yaml is a copy off https://raw.githubusercontent.com/rancher/local-path-provisioner/v0.0.28/deploy/local-path-storage.yaml
+    #assert_script_run( "curl " . data_url("aistack/$local_storage_name") . " -o $local_storage_name", 60);
+    assert_script_run("kubectl apply -f $local_path_url", timeout => 120);
+    assert_script_run("helm upgrade --install suse-private-ai private-ai-charts --namespace $namespace --create-namespace --values $vf_name --set open-webui.ingress.class=nginx", timeout => 600);
+    assert_script_run("kubectl get all --namespace $namespace");
+    sleep 180;
+
+    # Check pod status and log for successful
+    # Loop thru each pod and pod status is running,error,failed,CrashLoopBackoff,ContainerStatusUnknown skip to next pod
+    # any other pod status check the log for Failure,Error,Exception record the log and skip to next pod
+    # if not loop thru till it reaches the max_retries to ensure the pod comes to running or failure state.
+    # After reaching max_retries , record the pod details which does not run after reaching max_retries
+    my $max_retries = 15;
+    my $sleep_interval = 10;
+    my @out = split(' ', script_output("kubectl get pods --namespace $namespace -o custom-columns=':metadata.name'"));
+    record_info("Pod names", join(" ", @out));
+  POD_LOOP: foreach my $pod (@out) {
+        my $counter = 0;
+        my $start_time = time();
+        while ($counter++ < $max_retries) {
+            my $status = script_output("kubectl get pod $pod -n $namespace -o=jsonpath='{.status.phase}'", proceed_on_failure => 1);
+            my $logs = script_output("kubectl logs $pod -n $namespace", proceed_on_failure => 1);
+            if ($status eq 'Running') {
+                record_info("$pod is running. ");
+                next POD_LOOP;
+            } elsif ($status =~ /^(Error|Failed|CrashLoopBackOff|ContainerStatusUnknown)$/) {
+                record_info("$pod failed due to error in log: $logs \n ");
+                next POD_LOOP;
+            } else {
+                if ($logs =~ /ERROR|FAILURE|Exception|Failed/) {
+                    record_info("$pod failed due to error in log: $logs \n ");
+                    next POD_LOOP;
+                }    # if log
+                sleep $sleep_interval;
+            }    # if status
+        }    # while loop
```

Comment:
This logic is a bit odd, it would be good if after merging or even before you reeschedule the same test about 100 times:

[`openqa-cli -skip-chained-deps --repeat=50 --within-instance $job`](http://open.qa/docs/#_statistical_investigation)